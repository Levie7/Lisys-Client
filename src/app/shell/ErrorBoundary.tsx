import React from 'react';

interface ErrorBoundaryProps {
    onError: (error: Error) => void;
    renderOnError: React.FC;
}
export class ErrorBoundary extends React.Component<ErrorBoundaryProps, { hasError: boolean }> {
    constructor(props: ErrorBoundaryProps) {
        super(props);
        this.state = { hasError: false };
    }

    static getDerivedStateFromError(_: Error) {
        return { hasError: true };
    }

    componentDidCatch(error: Error) {
        this.props.onError(error);
    }

    render() {
        if (this.state.hasError) {
            return <this.props.renderOnError />;
        }
        return this.props.children;
    }
}
