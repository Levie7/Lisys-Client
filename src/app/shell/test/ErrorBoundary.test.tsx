import { shallow } from 'enzyme';
import React from 'react';

import { ErrorBoundary } from '../ErrorBoundary';

describe('ErrorBoundary', () => {
    const props = { onError: jest.fn(), renderOnError: jest.fn() };
    const wrapError = shallow(
        <h1 id='error' key='1'>
            Hello
        </h1>
    );
    const wrap = shallow(<ErrorBoundary {...props}>{wrapError}</ErrorBoundary>);
    wrap.simulateError('error');

    it('<ErrorBoundary/> - should render errorBoundary component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });

    it('should setError', () => {
        wrap.setState({ hasError: true });
    });

    describe('when has error', () => {
        beforeEach(() => {
            wrap.setState({ hasError: true });
        });

        it('should render on error', () => {
            expect(wrapError.find('#error').exists()).toBeTruthy();
        });
    });
});
