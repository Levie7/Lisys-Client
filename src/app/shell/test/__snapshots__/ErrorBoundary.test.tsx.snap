// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary <ErrorBoundary/> - should render errorBoundary component correctly 1`] = `ShallowWrapper {}`;
